main/src/com/google/refine/model/recon/StandardReconConfig.java

```diff
@@ -220,8 +226,10 @@ public ReconType getReconType() {
 
     @Override
     @JsonIgnore
-    public int getBatchSize() {
-        return 10;
+    public int getBatchSize(int rowCount) {
+
+        return Math.min(Math.max(rowCount / 100, 10), batchSize);
+
```

Comment:
Cosmetic changes: I'd remove the blank lines before and after the return statement (not sure if our linter could take care of that…)